### PR TITLE
build: fix small browser test coverage

### PR DESCRIPTION
**Pull request by DevVersion**
* No longer just runs the tests against the first browser of a test platform (travis, saucelabs, browserstack)
* Stability is more valuable than speed (runs sequentially on CI now)
* Update Saucelabs Connect
* Remove unused spec reporter
* Update from Edge 14 to Edge 17
* Fix pixel deviations in Edge 17

### CODE REVIEW

File: src/lib/select/select.spec.ts [hunk 1]
```diff
@@ -2917,11 +2917,15 @@ describe('MatSelect', () => {
       const topDifference = Math.floor(optionTop) -
           Math.floor(triggerTop - triggerFontSize - triggerExtraLineSpaceAbove);
 
+
+      console.log({optionTop});
+      console.log({triggerTop}, ' - ', (triggerTop - triggerFontSize - triggerExtraLineSpaceAbove));
```
Comment (crisbeto) [inline src/lib/select/select.spec.ts]: Leftover logs?

File: src/lib/select/select.spec.ts [hunk 2]
```diff
@@ -2917,11 +2917,15 @@ describe('MatSelect', () => {
...
+
       // Expect the coordinates to be within a pixel of each other. We can't rely on comparing
       // the exact value, because different browsers report the various sizes with slight (< 1px)
       // deviations.
-      expect(Math.abs(topDifference) < 2)
-          .toBe(true, `Expected trigger to align with option ${index}.`);
+      expect(Math.abs(topDifference))
+          .toBeLessThan(2, `Expected trigger to align with option ${index}.`);
```
Comment (crisbeto) [inline src/lib/select/select.spec.ts]: The old expectation was intentional. Changing it to `toBeLessThan` will log out something like `Expected 7 to be less than 2, Expected trigger to align with option 2.` which can be confusing.

File: src/lib/select/select.spec.ts [hunk 3]
```diff
@@ -3356,7 +3360,7 @@ describe('MatSelect', () => {
         formField.style.marginRight = '20px';
       }));
 
-      it('should align the first option properly when scrolled', fakeAsync(() => {
+      fit('should align the first option properly when scrolled', fakeAsync(() => {
```
Comment (crisbeto) [inline src/lib/select/select.spec.ts]: Leftover `fit`.

File: src/lib/select/select.spec.ts (hunk 1 above)
Comment (DevVersion) [reply src/lib/select/select.spec.ts]: As mentioned on Slack. Debugging is happening right now. One specific select test is timing out on Edge Saucelabs.

**Comment by DevVersion** [comment]
Sorry for the commit spam on this PR, but this is just necessary when doing changes that need to be verified with CI builds.

Overall I've updated this PR **again** because Browserstack no longer supports running mobile devices with our current license. Should be still good to go.

**Comment by DevVersion** [comment]
Rebased.

**Pull request merged**